### PR TITLE
Add GUI support to DataFileNew

### PR DESCRIPTION
**Pull request by ashmaroli**
- [x] add GUI to `DataFileNew`
- [x] update spec file for view
- [x] lock GUI support to just YAML and JSON files
    - ~~**Either** listen to `InputPath` value and disable `createButton`,~~
    - **or** add a form component to **`select`** the file extension.
- [x] update documentation

### CODE REVIEW

**Comment by mertkahyaoglu** [comment]
There is one issue I encountered. When you change a field type in the gui mode, for example making it list, `Create` button is disabled. I think we should pass `|| fieldChanged` to create button's `active` prop.

File: src/containers/views/DataFileNew.js [hunk 1]
```diff
@@ -35,57 +55,123 @@ export class DataFileNew extends Component {
...
+  }
+
+  handleClickSave(e) {
     const { datafileChanged, putDataFile } = this.props;
+
+    // Prevent the default event from bubbling
+    preventDefault(e);
+
+    let filename;
     if (datafileChanged) {
```
Comment (mertkahyaoglu) [inline src/containers/views/DataFileNew.js]: What about `fieldChanged`? Save action won't be dispatched if users directly edit in the gui mode.

**Comment by ashmaroli** [comment]
> I think we should pass `|| fieldChanged` to create button's active prop.

No, that was intentional.. The `Create` button will be `activated` only if the `path` has been entered..

File: src/containers/views/DataFileNew.js (hunk 1 above)
Comment (ashmaroli) [inline src/containers/views/DataFileNew.js]: Its an interface for `Create`. Enter the path in `InputPath` and viola!

**Comment by mertkahyaoglu** [comment]
Could you try the following steps;

1. Click `New Data File`
2. Switch to GUI mode.
3. Enter filename as `test`
4. Select extension
5. Change first field type to list
6. Expect Create button to be disabled.

**Comment by ashmaroli** [comment]
Uh-oh.. didn't expect that bug..
Thanks.. will update :+1:

File: src/containers/views/DataFileNew.js (hunk 1 above)
Comment (mertkahyaoglu) [inline src/containers/views/DataFileNew.js]: Once you fix the below bug, you will face this one 😬

Comment (mertkahyaoglu) [inline src/containers/views/DataFileNew.js]: Create button is active but clicking it won't create anything.

Comment (ashmaroli) [inline src/containers/views/DataFileNew.js]: we're not sending a data object to the `put` action via the GUI, so this shouldn't matter..

**Comment by ashmaroli** [comment]
I've made a minor change in the save button behavior:  It'll get `disabled` when the GUI input value for the file's basename becomes `nil`.This is required because the GUI for this view would otherwise send a request with `filename: .yml` thereby bypassing the `Errors` component

File: src/containers/views/DataFileNew.js (hunk 1 above)
Comment (mertkahyaoglu) [inline src/containers/views/DataFileNew.js]: But I can't create the datafile 😄  I'm copy-pasting steps below;

1. Click New Data File
2. Switch to GUI mode.
3. Enter filename as test
4. Select extension
5. Change first field type to list
6. Add some array items.
7. Try to `Create` datafile.
8. Nothing happens.

Comment (ashmaroli) [inline src/containers/views/DataFileNew.js]: forgive my nearsightedness, Master.. :pray: 
pushed a patch..

**Comment by mertkahyaoglu** [comment]
Awesome 👍  Thanks @ashmaroli .

**Pull request merged**